deployment/helm/kueue/templates/visibility/service.yaml

```diff
@@ -0,0 +1,17 @@
+{{- if include "kueue.isFeatureGateEnabled" (dict "List" .Values.controllerManager.featureGates "Feature" "VisibilityOnDemand") }}
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+  {{- include "kueue.labels" . | nindent 4 }}
+  name: '{{ include "kueue.fullname" . }}-visibility-server'
+  namespace: '{{ .Release.Namespace }}'
+spec:
+  ports:
+    - name: https
+      port: 443
+      protocol: TCP
+      targetPort: 8082
```

Comment:
I'm unclear how this Visibility Server works.  This service will expose something on the default `https` port, but where does the 8082 target port come from?  Also, we typically use Traefik Endpoints for this, so is this different from that?

Reply:
> I'm unclear how this Visibility Server works. This service will expose something on the default `https` port, but where does the 8082 target port come from? Also, we typically use Traefik Endpoints for this, so is this different from that?

The Visibility service is an API endpoint used to monitor pending workloads on demand. As of the date of this comment, it is an alpha level feature and is turned off by default, including on our development cluster. It may be enabled by setting the feature gates configuration 'VisibilityOnDemand=true'

Details are available at https://kueue.sigs.k8s.io/docs/tasks/manage/monitor_pending_workloads/pending_workloads_on_demand/